### PR TITLE
Add skeleton OAuth, scaffolding for enabling TestFlow tests

### PR DESCRIPTION
**Pull request by daveta**
Sorry, a bunch of changes for enabling testflow and skeleton for new OAuth client.

### CODE REVIEW

**Comment by daveta** [comment]
Oh, and changed the version to the SNAPSHOT convention.  I'm going to upload these packages to maven central..

File: libraries/bot-connector/src/main/java/com/microsoft/bot/connector/ConnectorClientFuture.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
+package com.microsoft.bot.connector;
+
+import com.microsoft.bot.connector.implementation.ConnectorClientImpl;
+import com.microsoft.rest.RestClient;
+import com.microsoft.rest.credentials.ServiceClientCredentials;
+
+public class ConnectorClientFuture extends ConnectorClientImpl {
```
Comment (johnataylor) [inline libraries/bot-connector/src/main/java/com/microsoft/bot/connector/ConnectorClientFuture.java]: "Impl" nice:-)

File: libraries/bot-connector/src/main/java/com/microsoft/bot/connector/authentication/OAuthClient.java [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+        if (StringUtils.isEmpty(userId)) {
+            throw new IllegalArgumentException("userId");
+        }
+        if (StringUtils.isEmpty(connectionName)) {
+            throw new IllegalArgumentException("connectionName");
+        }
+
+        //boolean shouldTrace = // ServiceClientTracing.IsEnabled;
+        String invocationId = null;
+//        if (shouldTrace)
```
Comment (johnataylor) [inline libraries/bot-connector/src/main/java/com/microsoft/bot/connector/authentication/OAuthClient.java]: presumably a work in progress??? hence the review title "skeleton"

File: libraries/botbuilder-schema/src/main/java/com/microsoft/bot/schema/models/MessageActivity.java [hunk 3]
```diff
@@ -143,9 +143,9 @@ void setAttachments(List<Attachment> attachments) {
     /// <summary>
     /// Create an instance of the Activity class with IConversationUpdateActivity masking
     /// </summary>
-    public static ConversationUpdateActivity CreateConversationUpdateActivity()
+    public static MessageActivity CreateConversationUpdateActivity()
     {
-        return (ConversationUpdateActivity) new MessageActivity()
+        return (MessageActivity) new MessageActivity()
```
Comment (johnataylor) [inline libraries/botbuilder-schema/src/main/java/com/microsoft/bot/schema/models/MessageActivity.java]: consider writing this without the cast (which can throw) - instead use a local variable.

File: libraries/botbuilder/src/main/java/com/microsoft/bot/builder/core/BotAdapter.java [hunk 4]
```diff
@@ -69,7 +69,7 @@ public BotAdapter Use(Middleware middleware) {
     /// an array of <see cref="ResourceResponse"/> objects containing the IDs that
     /// the receiving channel assigned to the activities.</remarks>
     /// <seealso cref="ITurnContext.OnSendActivities(SendActivitiesHandler)"/>
-    public abstract CompletableFuture<ResourceResponse[]> SendActivities(TurnContext context, Activity[] activities) throws InterruptedException;
+    public abstract CompletableFuture<ResourceResponse[]> SendActivities(TurnContext context, Activity[] activities) throws InterruptedException, ServiceKeyAlreadyRegisteredException;
```
Comment (johnataylor) [inline libraries/botbuilder/src/main/java/com/microsoft/bot/builder/core/BotAdapter.java]: heads up - using a custom exception does reflect the current state of the C# however that is not best practice and we will be updating the c#

File: libraries/bot-connector/src/main/java/com/microsoft/bot/connector/authentication/SimpleCredentialProvider.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+    public SimpleCredentialProvider() {
+    }
+    public SimpleCredentialProvider(String appId, String password) {
+        this._appId = appId;
+        this._password = password;
+    }
+
+    public String getAppId() {
+        return _appId;
```
Comment (johnataylor) [inline libraries/bot-connector/src/main/java/com/microsoft/bot/connector/authentication/SimpleCredentialProvider.java]: what is the convention around "this" and "_" using both seems a little excessive?
presumably we can drop the use of "this"
(realize the C# is inconsistent but we'll be fixing that)

File: libraries/botbuilder/src/test/java/com/microsoft/bot/builder/core/adapters/TestFlow.java [hunk 6]
```diff
@@ -2,18 +2,21 @@
...
 import java.lang.management.ManagementFactory;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
 import java.util.function.Function;
 
 import static java.util.concurrent.CompletableFuture.completedFuture;
+import static java.util.concurrent.CompletableFuture.supplyAsync;
 
 public class TestFlow {
     final TestAdapter adapter;
```
Comment (johnataylor) [inline libraries/botbuilder/src/test/java/com/microsoft/bot/builder/core/adapters/TestFlow.java]: should this be "_adapter" ?
(and consistently across the other member variables? and then not use the "this")

File: libraries/bot-connector/src/main/java/com/microsoft/bot/connector/authentication/OAuthClient.java (hunk 2 above)
Comment (daveta) [reply libraries/bot-connector/src/main/java/com/microsoft/bot/connector/authentication/OAuthClient.java]: Yes, much more to convert before I can fully light this up

File: libraries/botbuilder/src/test/java/com/microsoft/bot/builder/core/adapters/TestFlow.java (hunk 6 above)
Comment (daveta) [reply libraries/botbuilder/src/test/java/com/microsoft/bot/builder/core/adapters/TestFlow.java]: Apparently, no _ and accessing via "this."

File: libraries/botbuilder/src/main/java/com/microsoft/bot/builder/core/BotAdapter.java (hunk 4 above)
Comment (daveta) [reply libraries/botbuilder/src/main/java/com/microsoft/bot/builder/core/BotAdapter.java]: Waiting with anticipation..

File: libraries/bot-connector/src/main/java/com/microsoft/bot/connector/ConnectorClientFuture.java (hunk 1 above)
Comment (daveta) [reply libraries/bot-connector/src/main/java/com/microsoft/bot/connector/ConnectorClientFuture.java]: Heh.

**Pull request merged**